### ISSUE TITLE
Car Detection from Drones

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#235]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Car Detection from Drones</br>
:red_circle: **Aim** : Create a model which will identify cars from the drones footage.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/pear2jam/cars-drone-detection</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by manu31shukla** [comment]
Full Name: Manu Shukla
GitHub Profile Link: https://github.com/manu31shukla
Email ID: [[manu31shukla@gmail.com](mailto:manu31shukla@gmail.com)]

Approach for this project: 
I will perform an in-depth exploratory data analysis on the car detection dataset from drone footage and visualize the dataset to gain insights into the distribution of car images, background noise, and potential challenges.
By exploring the variations in lighting, angles, and perspectives in the images,  cleaning the dataset by removing any corrupted or incomplete images to ensure high data quality.
To increase the diversity of the dataset, I will apply data augmentation techniques such as rotation, scaling, and flipping.
Normalizing the pixel values and resizing the images to a consistent input size will be part of my preprocessing pipeline.
Then I will select 3-4 deep learning algorithms suitable for object detection, specifically for car detection.
Using popular deep learning frameworks such as TensorFlow or PyTorch, I will implement the chosen models.
I will fine-tune the model architectures based on the dataset characteristics and experiment with hyperparameters to optimize performance .Measuring relevant metrics such as precision, recall, and F1 score, I will compare the effectiveness of the different algorithms.


What is your participant role: Participant in SSOC 

Kindly assign the issue to me.

**Comment by abhisheks008** [comment]
Hi @manu31shukla nice to have you here. I like your approach for solving this issue. This issue will be assigned to you once the program starts officially.

Till them, star the project repository and explore more in open source.

**Comment by abhisheks008** [comment]
Issue assigned to @manu31shukla

**Comment by kanishkakataria** [comment]
Full Name: Kanishka Kataria
GitHub Profile Link: https://github.com/kanishkakataria
Email :[(kanishka.kataria99@gmail.com)]

Approach for this project:
I will perform an in-depth exploratory data analysis on the car detection dataset from drone footage and visualize the dataset to gain insights into the distribution of car images, background noise, and potential challenges.
I will augment the images, resize the images if necesary ,flip and rotate to train the models with more image train  data.

Normalizing pixel values will be covered under preprocessing.

Then I will select 3-4 deep learning algorithms suitable for object detection.

A comparative study of all the models like yolo, cnn, rcnn, resnet 50, vggnet ,detr and other models

Hyper tuning will be done for better performance of the models.Measuring relevant metrics such as precision, recall, and F1 score, I will compare the effectiveness of the different algorithms.

What is your participant role: contributor SSOC

Kindly assign the issue to me.

**Comment by abhisheks008** [comment]
> Full Name: Kanishka Kataria GitHub Profile Link: https://github.com/kanishkakataria Email :[([kanishka.kataria99@gmail.com](mailto:kanishka.kataria99@gmail.com))]
> 
> Approach for this project: I will perform an in-depth exploratory data analysis on the car detection dataset from drone footage and visualize the dataset to gain insights into the distribution of car images, background noise, and potential challenges. I will augment the images, resize the images if necesary ,flip and rotate to train the models with more image train data.
> 
> Normalizing pixel values will be covered under preprocessing.
> 
> Then I will select 3-4 deep learning algorithms suitable for object detection.
> 
> A comparative study of all the models like yolo, cnn, rcnn, resnet 50, vggnet ,detr and other models
> 
> Hyper tuning will be done for better performance of the models.Measuring relevant metrics such as precision, recall, and F1 score, I will compare the effectiveness of the different algorithms.
> 
> What is your participant role: contributor SSOC
> 
> Kindly assign the issue to me.

This issue is already assigned to a contributor. You can take other issues present here or you can create your own issues.

**Comment by Gaurav-576** [comment]
✅ To be Mentioned while taking the issue :

Full name : Gaurav Kumar SIngh
GitHub Profile Link : https://github.com/Gaurav-576
Email ID : [gauravsingh96753@gmail.com](mailto:gauravsingh96753@gmail.com)
Participant ID (if applicable):
Approach for this Project :
I would like to perform web scrapping to add more images to the dataset. Using data augmentation to create more data for the model to train onto. Then I would be using CNN to identify whether it is a tiger or not using Tensorflow and keras. Improving the overall accuracy of the model and focusing on False Negative recall of the confusion matrix. For real-time object detection, Yolo can be used along with OpenCV which I would like to discuss with the mentors. 
What is your participant role? GSSoC'24

Thus, I request the project admins to assign this issue to me.

**Comment by Gaurav-576** [comment]
The other models which can be used includes the Tensorflow Object Detection, dynamic computation graph by PyTorch and for mobile devices, mobilenet can be used.

**Comment by abhisheks008** [comment]
Hi @Gaurav-576 thanks for the comment. But you can work on one issue at a time.

**Comment by Adhivp** [comment]
I would like to work on this @abhisheks008

**Comment by abhisheks008** [comment]
Please share your approach in a brief way as mentioned in the issue template.
@Adhivp

**Comment by Gaurav-576** [comment]
✅ To be Mentioned while taking the issue :

Full name : Gaurav Kumar SIngh
GitHub Profile Link : https://github.com/Gaurav-576
Email ID : [gauravsingh96753@gmail.com](mailto:gauravsingh96753@gmail.com)
Participant ID (if applicable):
Approach for this Project :
I would like to perform web scrapping to add more images to the dataset. Using data augmentation to create more data for the model to train onto. Then I would be using CNN to identify whether it is a tiger or not using Tensorflow and keras. Improving the overall accuracy of the model and focusing on False Negative recall of the confusion matrix. For real-time object detection, Yolo can be used along with OpenCV which I would like to discuss with the mentors.
What is your participant role? GSSoC'24

Thus, I request the project admins to assign this issue to me.

**Comment by abhisheks008** [comment]
> ✅ To be Mentioned while taking the issue :
> 
> Full name : Gaurav Kumar Singh GitHub Profile Link : https://github.com/Gaurav-576 Email ID : [gauravsingh96753@gmail.com](mailto:gauravsingh96753@gmail.com) Participant ID (if applicable): Approach for this Project : I would like to perform web scrapping to add more images to the dataset. Using data augmentation to create more data for the model to train onto. Then I would be using CNN to identify whether it is a tiger or not using Tensorflow and keras. Improving the overall accuracy of the model and focusing on False Negative recall of the confusion matrix. For real-time object detection, Yolo can be used along with OpenCV which I would like to discuss with the mentors. What is your participant role? GSSoC'24
> 
> Thus, I request the project admins to assign this issue to me.

I think you haven't understood the problem statement correctly, it's a car detection project not a tiger detection.

Also mention 3-4 models for this problem statement which you are going to implement for this dataset.

**Comment by Gaurav-576** [comment]
✅ To be Mentioned while taking the issue :

Full name : Gaurav Kumar Singh
GitHub Profile Link : https://github.com/Gaurav-576
Email ID : [gauravsingh96753@gmail.com](mailto:gauravsingh96753@gmail.com)
Participant ID (if applicable):
Approach for this Project :
I would like to perform web scrapping to add more images to the dataset. Using data augmentation to create more data for the model to train onto. Then I would be using CNN to identify whether it is a Car or not using Tensorflow and keras. Improving the overall accuracy of the model and focusing on False Negative recall of the confusion matrix. For real-time object detection, Yolo can be used along with OpenCV which I would like to discuss with the mentors.


I would be using a Convolutional Neural Network (CNN) which would perfectly fir with my dataset and generate me proper results as per my requirements of the project.


The other models I can use include transfer learning techniques such as YOLO, ImageNet, SSD, etc.
What is your participant role? GSSoC'24

Thus, I request the project admins to assign this issue to me.

**Comment by abhisheks008** [comment]
Implement more than 4 models to get this issue as level 3.

Assigning this issue to you @Gaurav-576

**Comment by Gaurav-576** [comment]
Great, thanks for assigning this issue to me and I'll do my best and try to implement more than 3 models for the problem statement with high accuracy AI models.
